src/utils/components/AddBootableVolumeModal/components/VolumeSource/components/RegistrySource.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FC, useEffect } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import ContainerSource from '@catalog/templatescatalog/components/TemplatesCatalogDrawer/StorageSection/CustomizeSource/Sources/ContainerSource';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+
+import { AddBootableVolumeState, SetBootableVolumeFieldType } from '../../../utils/constants';
+
+type RegistrySourceProps = {
+  bootableVolume: AddBootableVolumeState;
+  setBootableVolumeField: SetBootableVolumeFieldType;
+};
+
+const RegistrySource: FC<RegistrySourceProps> = ({ bootableVolume, setBootableVolumeField }) => {
+  const { t } = useKubevirtTranslation();
+  const { registryCredentials, registryURL } = bootableVolume || {};
+
+  const methods = useForm({
+    defaultValues: {
+      'volume-registry-containerImage': registryURL || '',
+      'volume-registry-password': registryCredentials?.password || '',
+      'volume-registry-username': registryCredentials?.username || '',
+    },
+    mode: 'all',
+  });
```

Comment:
All the form states are in `bootableVolume`, and I don't want to add a new form management tool here.
You can create a function to change the password and other things in the `bootableVolume` state using the function `setBootableVolumeField('password')`